trieste/acquisition/sampler.py

```diff
@@ -19,49 +19,16 @@
 from __future__ import annotations
 
 from abc import ABC, abstractmethod
-from typing import Callable
 
 import tensorflow as tf
 import tensorflow_probability as tfp
-from gpflux.layers.basis_functions import RandomFourierFeatures
 from scipy.optimize import bisect
 
-from ..data import Dataset
 from ..models import ProbabilisticModel
 from ..types import TensorType
-from ..utils import DEFAULTS
 
 
-class Sampler(ABC):
-    r"""
-    A :class:`Sampler` samples a specific quantity across a discrete set of points
-    according to an underlying :class:`ProbabilisticModel`.
-    """
-
-    def __init__(self, sample_size: int, model: ProbabilisticModel):
-        """
-        :param sample_size: The desired number of samples.
-        :param model: The model to sample from.
-        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive.
-        """
-        tf.debugging.assert_positive(sample_size)
-
-        self._sample_size = sample_size
-        self._model = model
-
-    def __repr__(self) -> str:
-        """"""
-        return f"{self.__class__.__name__}({self._sample_size!r}, {self._model!r})"
-
-    @abstractmethod
-    def sample(self, at: TensorType) -> TensorType:
-        """
-        :param at: Input points that define the sampler.
-        :return: Samples.
-        """
-
-
-class ThompsonSampler(Sampler, ABC):
+class ThompsonSampler(ABC):
```

Comment:
It's a bit confusing that the Thompson samplers are here and the others are all now under model. Is there a reason why these can't be moved too (even though they're not generated by the model)?

Reply:
The Thompson samplers are defined for any model and are used just in our acquisition functions. I think it makes sense to keep them here. My next PR, where I tidy up the Thompson sampling rule will make this neater

Reply:
I'm fine with the location of these, but it's perhaps a bit strange that these two quite different samplers will subclass the same `ThompsonSampler` class. Is this the sort of thing you're thinking of tidying?

Reply:
indeed

Reply:
My first impression is that this is the right call. Thompson samplers are different in nature because we assume a particular usage of the model (e.g. it is an objective and we want to minimise it), while the rest is a modelling thing (how do you get a sample from the joint distribution of your model).